taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -1110,7 +1141,22 @@ void CodeGenLLVM::visit(ReturnStmt *stmt) {
 
 void CodeGenLLVM::visit(LocalLoadStmt *stmt) {
   TI_ASSERT(stmt->width() == 1);
-  llvm_val[stmt] = builder->CreateLoad(llvm_val[stmt->src[0].var]);
+#ifdef TI_LLVM_15
+  // FIXME: get ptr_ty from taichi instead of llvm.
+  llvm::Type *ptr_ty = nullptr;
+  auto *val = llvm_val[stmt->src[0].var];
+  if (auto *alloc = llvm::dyn_cast<llvm::AllocaInst>(val))
+    ptr_ty = alloc->getAllocatedType();
+  if (!ptr_ty && stmt->src[0].var->element_type().is_pointer()) {
+    ptr_ty = llvm_type(stmt->src[0].var->element_type().ptr_removed());
+  }
+  TI_ASSERT(ptr_ty);
+#endif
+  llvm_val[stmt] = builder->CreateLoad(
+#ifdef TI_LLVM_15
```

Comment:
We probably wanna reduce the use of TI_LLVM_15 macro in the same scope. Can we join these two TI_LLVM_15 macros?